packages/salesforcedx-vscode-core/src/commands/auth/forceAuthLogout.ts

```diff
@@ -47,3 +58,77 @@ const commandlet = new SfdxCommandlet(
 export async function forceAuthLogoutAll() {
   await commandlet.run();
 }
+
+export class AuthLogoutDefault extends LibraryCommandletExecutor<string> {
+  constructor() {
+    super(
+      nls.localize('force_auth_logout_default_text'),
+      'force_auth_logout_default',
+      OUTPUT_CHANNEL
+    );
+  }
+
+  public async run(
+    response: ContinueResponse<string>,
+    progress?: Progress<{
+      message?: string | undefined;
+      increment?: number | undefined;
+    }>,
+    token?: CancellationToken
+  ): Promise<boolean> {
+    try {
+      await removeUsername(response.data);
+    } catch (e) {
+      telemetryService.sendException(e.name, e.message);
+      return false;
+    }
+    return true;
+  }
+}
+
+export async function forceAuthLogoutDefault() {
+  const { username, isScratch, error } = await resolveDefaultUsername();
+  if (error) {
+    telemetryService.sendException(error.name, error.message);
+    notificationService.showErrorMessage('Logout failed to run');
+  } else if (username) {
+    const logoutCommandlet = new SfdxCommandlet(
+      new SfdxWorkspaceChecker(),
+      isScratch
+        ? new ScratchOrgLogoutParamsGatherer(username)
+        : new SimpleGatherer<string>(username),
+      new AuthLogoutDefault()
+    );
+    await logoutCommandlet.run();
+  } else {
+    notificationService.showInformationMessage(
+      nls.localize('auth_logout_no_default_org')
+    );
+  }
+}
+
+async function removeUsername(username: string) {
+  await forceConfigSet('');
```

Comment:
a general musing - can we eventually replace the call to the `sfdx force:config:set` command with a direct call to `@salesforce/core`?

Reply:
@AnanyaJha Yes, I was planning to convert the spike story into a story to do that. I felt it was out of scope for this work.